### PR TITLE
Fixes #21390: Broken plugins menu interface

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/21390

### CODE REVIEW

**Comment by amousset** [comment]
PR updated with a new commit

File: branding/src/main/scala/com/normation/plugins/branding/BrandingPluginDef.scala [hunk 1]
```diff
@@ -61,16 +60,16 @@ class BrandingPluginDef(override val status: PluginStatus) extends DefaultPlugin
...
+    ((Menu("640-brandingManagement", <span>Branding</span>) /
         "secure" / "administration" / "brandingManagement"
         >> LocGroup("administrationGroup")
         >> TestAccess ( () => Boot.userIsAllowed("/secure/administration/policyServerManagement", Administration.Read) )
         >> Template(() => ClasspathTemplates( "template" :: "brandingManagement" :: Nil ) openOr <div>Template not found</div>)
-    )
+    ).toMenu, None) :: Nil
   }
 
-  override def pluginMenuParent: Option[Menu] = {
```
Comment (VinceMacBuche) [inline branding/src/main/scala/com/normation/plugins/branding/BrandingPluginDef.scala:73]: You don't need to override it anymore

File: branding/src/main/scala/com/normation/plugins/branding/BrandingPluginDef.scala [hunk 2]
```diff
@@ -61,16 +60,16 @@ class BrandingPluginDef(override val status: PluginStatus) extends DefaultPlugin
...
 
-  override def pluginMenuEntry: Option[Menu] = {
-    Some(Menu("640-brandingManagement", <span>Branding</span>) /
+  override def pluginMenuEntry: List[(Menu, Option[String])] = {
+    ((Menu("640-brandingManagement", <span>Branding</span>) /
         "secure" / "administration" / "brandingManagement"
         >> LocGroup("administrationGroup")
         >> TestAccess ( () => Boot.userIsAllowed("/secure/administration/policyServerManagement", Administration.Read) )
         >> Template(() => ClasspathTemplates( "template" :: "brandingManagement" :: Nil ) openOr <div>Template not found</div>)
-    )
```
Comment (VinceMacBuche) [inline branding/src/main/scala/com/normation/plugins/branding/BrandingPluginDef.scala:70]: None should be "Some(MenuUtils.utilitiesMenu)"

**Review by VinceMacBuche — CHANGES_REQUESTED** [review]
The semantics of the two function changed.

pluginMenuParent, become a list of new menu parents created by the plugin, but if you were using an existing menu before you don't need to override it

pluginMenuEntry, is the list of the new pages and they optionnal parent menu (None => goes to plugins, Some(id-menu) to put it in menu of id id-menu

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Review by VinceMacBuche — APPROVED** [review]

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**